tools/gradle-plugin/src/main/kotlin/io/github/composegears/valkyrie/gradle/ValkyrieGradlePlugin.kt

```diff
@@ -0,0 +1,70 @@
+package io.github.composegears.valkyrie.gradle
+
+import io.github.composegears.valkyrie.generator.jvm.imagevector.OutputFormat
+import io.github.composegears.valkyrie.generator.jvm.imagevector.PreviewAnnotationType
+import io.github.composegears.valkyrie.gradle.GenerateImageVectorsTask.Companion.TASK_NAME
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.dsl.KotlinAndroidProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSetContainer
+import org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompile
+
+@Suppress("unused") // Registered as a Gradle plugin.
+class ValkyrieGradlePlugin : Plugin<Project> {
+    override fun apply(target: Project): Unit = with(target) {
+        val extension = extensions.create("valkyrie", ValkyrieExtension::class.java).apply {
+            // Using the same defaults as `SvgXmlToImageVectorCommand` in tools/cli.
+            packageName.convention(packageNameOrThrow())
+            generateAtSync.convention(false)
+            outputDirectory.convention(layout.buildDirectory.dir("generated/sources/valkyrie"))
+            outputFormat.convention(OutputFormat.BackingProperty)
+            useComposeColors.convention(true)
+            generatePreview.convention(false)
+            previewAnnotationType.convention(PreviewAnnotationType.AndroidX)
+            useFlatPackage.convention(false)
+            useExplicitMode.convention(false)
+            addTrailingComma.convention(false)
+            indentSize.convention(4)
+        }
+
+        pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+            registerTasks<KotlinJvmProjectExtension>(extension)
+        }
+
+        pluginManager.withPlugin("org.jetbrains.kotlin.multiplatform") {
+            registerTasks<KotlinMultiplatformExtension>(extension)
+        }
+
+        pluginManager.withPlugin("org.jetbrains.kotlin.android") {
+            pluginManager.withPlugin("com.android.base") {
+                registerTasks<KotlinAndroidProjectExtension>(extension)
+            }
+        }
+
+        val codegenTasks = tasks.withType(GenerateImageVectorsTask::class.java)
+
+        // Run generation immediately if we're syncing Intellij/Android Studio - helps to speed up dev cycle
+        val isIdeSyncing = System.getProperty("idea.sync.active") == "true"
+        if (extension.generateAtSync.getOrElse(false) && isIdeSyncing) {
```

Comment:
The value of `extension.generateAtSync` could be read properly here? It's always `false`. I believe you need to add a test for this and read the value in `afterEvaluate`.